src/test/java/com/alkemy/ong/controller/UserControllerTest.java

```diff
@@ -231,21 +226,91 @@ void TestDeleteUserById_Admin_Forbidden() throws Exception {
-    void testUpdate() throws Exception {
+    void testUpdateUserById_VerbPatch() throws Exception {
 
+        UserDto dtoUpdate = new UserDto();
+        dtoUpdate.setId("123456");
+        dtoUpdate.setFirstName("lola");
+        dtoUpdate.setLastName("bunny");
+        dtoUpdate.setEmail("demo@email.com");
+        dtoUpdate.setPhoto("default.jpg");
 
+        Mockito.when(userRepository
+                .save(userEntity)).thenReturn(userEntity);
+        UserEntity userEntityUpdate = new UserEntity();
+        userEntityUpdate.setId("123456");
+        userEntityUpdate.setFirstName("lola");
+        userEntityUpdate.setLastName("bunny");
+        userEntityUpdate.setEmail("demo@email.com");
+        userEntityUpdate.setPhoto("default.jpg");
+        Mockito.when(userRepository
+                        .findById(Mockito.anyString()))
+                .thenReturn(Optional.of(userEntity));
+        Mockito.when(userRepository
+                .save(userEntityUpdate)).thenReturn(userEntityUpdate);
+            String updatedContent=objectWriter.writeValueAsString(userEntityUpdate);
+            MockHttpServletRequestBuilder mockHttpRequest = patch("/users/123456",dtoUpdate)
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .accept(MediaType.APPLICATION_JSON)
+                    .content(updatedContent);
+            mockMvc.perform(mockHttpRequest).andExpect(status().isOk());
 
     }
 
-    }
-
-
-
-
-
+    @Test
+    @WithUserDetails(USER)
+    void testUpdateUserById_VerbPatchIsNotFound() throws Exception {
+        doThrow(new ParamNotFound("An error occurred")).when(this.userService).updateUser((String) any(),(UserDto) any());
+
+        UserDto dtoUpdate = new UserDto();
+        dtoUpdate.setFirstName("lola");
+        dtoUpdate.setLastName("bunny");
+        dtoUpdate.setEmail("demo@email.com");
+        dtoUpdate.setPhoto("default.jpg");
+
+        UserEntity userEntityUpdate = new UserEntity();
+        userEntityUpdate.setId("123456");
+        userEntityUpdate.setFirstName("lola");
+        userEntityUpdate.setLastName("bunny");
+        userEntityUpdate.setEmail("demo@email.com");
+        userEntityUpdate.setPhoto("default.jpg");
+        String updatedContent=objectWriter.writeValueAsString(userEntityUpdate);
+
+        MockHttpServletRequestBuilder requestBuilder = MockMvcRequestBuilders.patch("/users/{id}", "42",dtoUpdate)
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON)
+                .content(updatedContent);
+        mockMvc.perform(requestBuilder).andExpect(status().isNotFound());
 
+    }
 
+    @Test
+    @WithUserDetails(ADMIN)
+    void testUpdateAdminById_VerbForbidden() throws Exception {
+        doThrow(new ParamNotFound("An error occurred")).when(this.userService).updateUser((String) any(),(UserDto) any());
+
+        UserDto dtoUpdate = new UserDto();
+        dtoUpdate.setFirstName("lola");
```

Comment:
I suggest to implement a class in util package to reuse this lines

Reply:
I will investigate how to do it

Reply:
I tried but the code gives me an error, I still refactored the code